lib/src/main/java/io/ably/lib/transport/WebSocketTransport.java

```diff
@@ -234,6 +258,13 @@ public void onError(final Exception e) {
             connectListener.onTransportUnavailable(WebSocketTransport.this, new ErrorInfo(e.getMessage(), 503, 80000));
         }
 
+        @Override
+        protected void onSetSSLParameters(SSLParameters sslParameters) {
```

Comment:
There's a niggling worry in the back of my head that we should perhaps consider making this more sophisticated, such that it will prefer the default implementation if we have no reason to expect it to fail... I recognise this is `lib/` code, but is there any way that we can detect if the `setEndpointIdentificationAlgorithm` method is available and call `super`'s implementation if it is? Presumably the subset of runtime devices without it is quite small. 🤔

Reply:
We can make it more general (as this is an old Java API version problem rather than an Android-specific problem) and call the `super.onSetSSLParameters()` in a `try / catch` block and try to catch the `NoSuchMethodError` if it is thrown :thinking: Then, if it is thrown we can send a warning (or info perhaps) to the log handler and continue, as the verification will happen in the `isHostnameVerified()` method. WDYT?
```java
try {
    super.onSetSSLParameters(sslParameters);
} catch (NoSuchMethodError exception) {
    Log.w(TAG, "Error when trying to set SSL parameters, most likely due to an old Java API version", exception);
}
```

Reply:
I like that. Furthermore, could you enhance that to include some state that is set if the method call was successful, so that `isHostnameVerified()` can be skipped later on in the lifecycle?

Reply:
Good idea, here's the change :wink:  https://github.com/ably/ably-java/pull/808/commits/f35c5b6a11956637a4ed0a9eeebc554693ff8778